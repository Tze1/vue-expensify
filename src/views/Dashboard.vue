<template>
  <div v-if="userLoggedIn" class="dashboard component">
    <h2 class="dashboard-title">Dashboard</h2>
    <ExpenseListFilters :filters="filters" />
    <ExpenseList />
  </div>
</template>

<script>
  import { mapState } from 'vuex';
  import ExpenseListFilters from '../components/ExpenseListFilters';
  import ExpenseList from '../components/ExpenseList'

  export default {
    name: 'Dashboard',
    computed: mapState({
      userLoggedIn () {
        return this.$store.getters.userLoggedIn;
      },
      user: state => state.auth.user,
      filters: state => state.filters,
    }),
    components: {
      ExpenseListFilters,
      ExpenseList
    },
  };
</script>

<style scoped lang="scss">
  @import '../styles/app.scss';

  .dashboard.component {
    .dashboard-title {
      margin-bottom: 0;
    }
  }
</style>
