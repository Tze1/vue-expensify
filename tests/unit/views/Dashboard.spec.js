import { shallowMount } from '@vue/test-utils';
import Dashboard from '@/views/Dashboard';
import moment from 'moment';

const $store = {
  state: {
    auth: {
      user: {
        displayName: 'Test User',
        photoUrl: '/img/avatar-default.png',
      },
    },
    filters: {
      text: '',
      startDate: moment().subtract(30, 'days'),
      endDate: undefined,
      sortBy: 'date',
    },
  },
  getters: {
    userLoggedIn: state => {
      return (state.user !== undefined);
    },
  },
};
const wrapper = shallowMount(Dashboard, {
  mocks: { $store },
});

describe('Dashboard.vue', () => {
  it('renders Dashboard view properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
