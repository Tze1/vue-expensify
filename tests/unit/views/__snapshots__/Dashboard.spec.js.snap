// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard.vue renders Dashboard view properly 1`] = `
<div class="dashboard component">
  <h2 class="dashboard-title">Dashboard</h2>
  <expenselistfilters-stub filters="[object Object]"></expenselistfilters-stub>
  <expenselist-stub></expenselist-stub>
</div>
`;
